### ISSUE TITLE
Black border around an alpha image.

### ISSUE THREAD

**Issue by ExeRhythm** [MetalPetal/MetalPetal#235]
### Checklist
- [x] I've read the [README](https://github.com/MetalPetal/MetalPetal/blob/master/README.md)
- [x] I've searched for [existing GitHub issues](https://github.com/MetalPetal/MetalPetal/issues)
I have a question about alpha handling. If I overlay an image with a transparent layer I get some weird effect around the layer (black border).
![Screen Shot 2021-03-25 at 12 54 09](https://user-images.githubusercontent.com/52459150/112458574-fbc6e600-8d6d-11eb-8a0d-07b00626d88f.png)
<img width="443" alt="Screen Shot 2021-03-25 at 13 34 59" src="https://user-images.githubusercontent.com/52459150/112459492-f322df80-8d6e-11eb-8dd9-a68f534a3720.png">

(Apple logo is the layer)
My code:
```
func overlay(_ img1cg:CGImage, with img2cg:CGImage, scaling: Double, alpha: Double) -> CGImage? {
        let img1 = MTIImage(cgImage: img1cg, isOpaque: false)
        let img2 = MTIImage(cgImage: img2cg, isOpaque: false).premultiplyingAlpha()
        
        // Watermark Layer
        let aspectWatermark = Double(img2cg.height) / Double(img2cg.width)
        
        let watermarkWidth = Int(Double(img1cg.width) * scaling)
        let watermarkHeight = Int(scaling * Double(img1cg.width) * Double(aspectWatermark))
        let layer = MTILayer(content: img2, layoutUnit: .pixel, position: CGPoint(x: img1cg.width / 2, y: img1cg.height / 2), size: CGSize(width: watermarkWidth, height: watermarkHeight), rotation: 0, opacity: Float(alpha), blendMode: .normal)
        
        let filter = MTIMultilayerCompositingFilter()
        
        
        
        filter.inputBackgroundImage = img1
        filter.layers = [layer]
        
        guard let image = filter.outputImage else { return nil }
        
        let options = MTIContextOptions()
        guard let device = MTLCreateSystemDefaultDevice(), let context = try? MTIContext(device: device, options: options) else {
            return nil
        }

        do {
            let filteredImage = try context.makeCGImage(from: image)
            return filteredImage
        } catch {
            print(error)
        }
        return nil
    }
```
Does anybody know how I can fix this? Does this have anything to do with alpha type? Thanks in advance!

**Comment by YuAo** [comment]
Hi @ExeRhythm  `premultiplyingAlpha()` should do the tick. This looks like a bug of the CGImage loading system. I'll take a look asap.

**Comment by ExeRhythm** [comment]
Hi, @YuAo. Thanks for the response! I will be waiting.

**Comment by YuAo** [comment]
This is indeed a bug of the CGImage loading system. We should let the image loader to decide the alphaType of the loaded image instead of using the alpha type info in the input CGImage.

We did not notice this when testing on iOS because `ImageIO.framework` on iOS prefers premultiplied alpha, and our GCImage loader also loads the image to metal texture with premultiplied alpha. However, when loading a png image on macOS using ImageIO, the loaded CGImage trends to have straight alpha.

This is fixed via c0d11a8

@ExeRhythm  Can you give it a try?

**Comment by ExeRhythm** [comment]
Sure, will test it out right now

**Comment by ExeRhythm** [comment]
It works! Thank you very much for helping me!
<img width="610" alt="Screen Shot 2021-03-25 at 15 38 59" src="https://user-images.githubusercontent.com/52459150/112474137-4b161200-8d80-11eb-9a86-0ced2239f652.png">

**Comment by YuAo** [comment]
Thanks for reporting the bug :)

**Comment by YuAo** [comment]
This bug fix is included in the latest release - 1.17.0
